app/routes/_layout+/_layout.tsx

```diff
@@ -148,6 +150,14 @@ export default function App() {
               )}
             </div>
             <Toaster />
+            <ClientOnly fallback={null}>
+              {() =>
+                window.matchMedia("(max-width: 640px)").matches &&
```

Comment:
Lets move this check into a function with a clear name, so its more obvious what it does.